File: Leetcode Js/L_Final_value_of_variable_after_performing_operations.js
// Final Value of Variable After Performing Operations
// Array String

/**
 * @param {string[]} operations
 * @return {number}
 */
var finalValueAfterOperations = function (operations) {
  var [i, counter] = [0, 0];
  for (i = 0; i < operations.length; i++) {
    if (operations[i] == "X++" || operations[i] == "++X") {
      counter += 1;
    }
    if (operations[i] == "X--" || operations[i] == "--X") {
      counter -= 1;
    }
  }
  return counter;
};

console.log(finalValueAfterOperations(["--X", "X++", "X++"]));
console.log(finalValueAfterOperations(["++X", "++X", "X++"]));
console.log(finalValueAfterOperations(["X++", "++X", "--X", "X--"]));

File: Leetcode/L_Final_Value_Of_Variable_After_Performing_Operations.py
# Final Value of Variable After Performing Operations
# Array String

class Solution:
    def finalValueAfterOperations(self, operations) -> int:
        counter = 0
        for i in range(0, len(operations)):
            if(operations[i] == "X++" or operations[i] == "++X"):
                counter += 1
            if(operations[i] == "X--" or operations[i] == "--X"):
                counter -= 1
        return counter


X = Solution()
print(X.finalValueAfterOperations(["--X", "X++", "X++"]))
print(X.finalValueAfterOperations(["++X", "++X", "X++"]))
print(X.finalValueAfterOperations(["X++", "++X", "--X", "X--"]))
